### PR TITLE
Qx unsafe writer

### PR DESCRIPTION
**Pull request by sergeyQx**
Refactored UnsafeWriter to cache the writer only on specifically marked threads. Whenr UnsafeWriter is used from a different thread its intial size reduced to 4KB. UnsafeWriterStatistics added.

### CODE REVIEW

File: rd-net/RdFramework/Impl/SocketWire.cs [hunk 1]
```diff
@@ -112,8 +112,8 @@ protected Base(string id, Lifetime lifetime, [NotNull] IScheduler scheduler) : b
       private Timer StartHeartbeat()
       {
         void OnTimedEvent(object sender, ElapsedEventArgs e)
-        {
-          Ping();
+        {
+          Scheduler.Queue(Ping);
```
Comment (knah) [inline rd-net/RdFramework/Impl/SocketWire.cs:116]: Using the default scheduler here may lead to wire being marked as potentially disconnected in cases of busy scheduler. With the old semantic pings were always sent regardless of what other threads were doing. Perhaps a ping scheduler should be added?

File: rd-net/Test.RdFramework/UnsafeWriterTest.cs [hunk 2]
```diff
@@ -1,85 +1,176 @@
...
     public void TestWriterIsReused02()
     {
-      UnsafeWriter firstWriter;
-      using (var cookie = UnsafeWriter.NewThreadLocalWriter())
+      UnsafeWriter firstWriter = null, secondWriter = null;
+      var thread = new Thread(() =>
       {
-        firstWriter = cookie.Writer;
-      }
+        UnsafeWriter.AllowUnsafeWriterCaching = true;
```
Comment (knah) [inline rd-net/Test.RdFramework/UnsafeWriterTest.cs:48]: Given how widespread try-finally with this being set to true is, perhaps a cookie is in order?

File: rd-net/RdFramework/Impl/SocketWire.cs (hunk 1 above)
Comment (sergeyQx) [reply rd-net/RdFramework/Impl/SocketWire.cs:116]: I just restored the behavior that existed before my changes. I introduced that Scheduler and then removed

File: rd-net/Test.RdFramework/UnsafeWriterTest.cs (hunk 2 above)
Comment (sergeyQx) [reply rd-net/Test.RdFramework/UnsafeWriterTest.cs:48]: I do not expect many usages of the AllowUnsafeWriterCaching API. Most of the usages are in tests. Only rare threads are expected to allow cahcing. 
Cookie is possible

**Pull request merged**